ParquetClassLibrary/Sandbox/Parquets/ParquetParent.cs

```diff
@@ -1,22 +1,196 @@
+
+        /// <summary>The set of values that are allowed for specific ParquetIDs.  Defined by child classes.</summary>
+        protected abstract Range<ParquetID> Bounds { get; }
+
+        /// <summary>Backing field for the ID property.</summary>
+        private ParquetID _id;
+
+        /// <summary>Unique identifier of the parquet.</summary>
+        [JsonProperty(PropertyName = "in_ID")]
+        public ParquetID ID
+        {
+            get
+            {
+                return _id;
+
+            }
+            protected set
+            {
+                if (KnownIDs.Contains(value))
+                {
+                    Error.Handle($"Tried to create duplicate parquet with ID {value}.");
+                }
+                else
+                {
+                    KnownIDs.Add(value);
+                    _id = value;
+                }
+            }
+        }
+
+        /// <summary>Player-facing name of the parquet.</summary>
+        [JsonProperty(PropertyName = "in_name")]
+        public string Name { get; private set; }
 
         /// <summary>
         /// If a <see cref="T:ParquetClassLibrary.Sandbox.BiomeMask"/> flag is set,
         /// this parquet helps generate the corresponding <see cref="T:ParquetClassLibrary.Sandbox.Biome"/>.
         /// </summary>
-        internal BiomeMask AddsToBiome { get; private protected set; }
+        [JsonProperty(PropertyName = "in_addsToBiome")]
+        public BiomeMask AddsToBiome { get; private set; }
+
+        #region Initialization
+        /// <summary>
+        /// Used by children of the <see cref="T:ParquetClassLibrary.Sandbox.Parquets.ParquetParent"/> class.
+        /// </summary>
+        /// <param name="in_ID">Unique identifier for the parquet.  Cannot be null.</param>
+        /// <param name="in_name">Player-friendly name of the parquet.  Cannot be null.</param>
+        /// <param name="in_addsToBiome">
+        /// A set of flags indicating which, if any, <see cref="T:ParquetClassLibrary.Sandbox.Biome"/>
+        /// this parquet helps to generate.
+        /// </param>
+        [JsonConstructor]
+        protected ParquetParent(ParquetID in_ID, string in_name, BiomeMask in_addsToBiome = BiomeMask.None)
+        {
+            if (!CheckBounds(in_ID))
+            {
+                throw new ArgumentOutOfRangeException(nameof(in_ID));
+            }
+
+            ID = in_ID;
+            Name = in_name ?? throw new ArgumentNullException(nameof(in_name));
+            AddsToBiome = in_addsToBiome;
+        }
+        #endregion
+
+        #region Bounds Checking
+        /// <summary>
+        /// Ensures that the given identifier lies withint a specified range.
+        /// </summary>
+        /// <param name="in_ID">An identifier.</param>
+        /// <returns><c>true</c>, if identifier is within bounds, <c>false</c> otherwise.</returns>
+        private bool CheckBounds(ParquetID in_ID)
+        {
+            var result = false;
```

Comment:
Range is a very new... thing (I'm not sure actually), but I'd be surprised if there wasn't a Range.Contains function or something.

Reply:
Oh, that's a custom class! You do have ContainsValue as a function in there, though, so maybe you could use that.

Reply:
Thanks, good catch once again!
Fixed in bfd5866.